src/rulesets/Base/fastmath_able.jl

```diff
@@ -164,14 +164,16 @@ let
         @scalar_rule x - y (true, -1)
         @scalar_rule x / y (one(x) / y, -(Ω / y))
         #log(complex(x)) is required so it gives correct complex answer for x<0
-        @scalar_rule(x ^ y,
-            (ifelse(iszero(x), zero(Ω), y * Ω / x), Ω * log(complex(x))),
-        )
+        @scalar_rule(x ^ y, (
+            ifelse(iszero(x), ifelse(isone(y), one(Ω), zero(Ω)), y * Ω / x),
+            Ω * log(complex(x)),
+        ))
         # x^y for x < 0 errors when y is not an integer, but then derivative wrt y
         # is undefined, so we adopt subgradient convention and set derivative to 0.
-        @scalar_rule(x::Real ^ y::Real,
-            (ifelse(iszero(x), zero(Ω), y * Ω / x), Ω * log(oftype(Ω, ifelse(x ≤ 0, one(x), x)))),
-        )
+        @scalar_rule(x::Real ^ y::Real, (
+            ifelse(iszero(x), ifelse(isone(y), one(Ω), zero(Ω)), y * Ω / x),
+            Ω * log(oftype(Ω, ifelse(x ≤ 0, one(x), x))),
+        ))
```

Comment:
Is this right? Applied to the example from https://github.com/FluxML/Zygote.jl/issues/1036 it still gives zero gradient:
```
julia> oldrule(0.0, 0.9)
(0.0, 0.0)

julia> newrule(0.0, 0.9)
(0.0, 0.0)

julia> oldrule(0.0 + eps(), 0.9)
(4.0532396646334465e15, -36.04365338911715)

julia> newrule(0.0 + eps(), 0.9)
(4.0532396646334465e15, -36.04365338911715)
```

Reply:
What is `newrule` and `oldrule` here?

Reply:
Edited, sorry. I just copy-pasted from the above diff, to quickly try things.

Reply:
I don't think this is necessarily well-defined, since the derivative of `x ^ (1 - ε)` should be `(1 - ε) / x ^ ε`, so the limit as `x -> 0` diverges.

Reply:
Sure. But giving some large positive number seems more useful than an error, or NaN, and silently giving zero seems just wrong. 

E.g. if you have this in some optimisation problem, and it would lower your loss to move away from the boundary x=0 to x=0.1, then getting zero gradient there is misleading.

Reply:
So should this be `Inf` instead?

Reply:
Maybe. As I just wrote in parallel here https://github.com/FluxML/Zygote.jl/issues/1036#issuecomment-889832592, maybe it would be better to regularise everything, for x=0 (in this one-sided case) always give the gradients for say x = eps(). This will change smoothly between the different cases.

Reply:
The `Ω / x` term is just `x^(y-1)`, which for `x=0` is 0 for all `y>1`, 1 for `y=1` and `Inf` for all `y<1`. I like the `eps` idea. It also smooths the approach to `0^0==1`. Not certain all the consequences it could have though.